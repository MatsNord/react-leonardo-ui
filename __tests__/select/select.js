import React from 'react';
import renderer from 'react-test-renderer';

import Select from '../../src/select';

test('Normal select renders correctly', () => {
  const tree = renderer.create(
    <Select value="Value" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Select with states renders correctly', () => {
  const tree = renderer.create(
    <Select value="Value" invalid disabled />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Select with modifiers renders correctly', () => {
  const tree = renderer.create(
    <Select value="Value" variant="inverse" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Select passes props correctly', () => {
  const onChange = jest.fn(() => {});

  const tree = renderer.create(
    <Select className="my-class" data-x="x" title="Title" onChange={onChange} />
  ).toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onChange();
  expect(onChange.mock.calls.length).toBe(1);
});
