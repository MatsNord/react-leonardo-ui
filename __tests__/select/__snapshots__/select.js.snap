exports[`test Normal select renders correctly 1`] = `
<select
  className="lui-select"
  value="Value" />
`;

exports[`test Select passes props correctly 1`] = `
<select
  className="lui-select my-class"
  data-x="x"
  onChange={[Function]}
  title="Title" />
`;

exports[`test Select with modifiers renders correctly 1`] = `
<select
  className="lui-select lui-select--inverse"
  value="Value" />
`;

exports[`test Select with states renders correctly 1`] = `
<select
  className="lui-select"
  disabled={true}
  invalid={true}
  value="Value" />
`;
